pkg/token/execCredentialPlugin.go

```diff
@@ -85,14 +85,20 @@ func (p *execCredentialPlugin) Do() error {
 				return fmt.Errorf("failed to get refresher: %s", err)
 			}
 			klog.V(5).Info("refresh token")
+			refreshToken := token.RefreshToken
 			token, err := refresher.Token()
+
 			// if refresh fails, we will login using token provider
 			if err != nil {
 				klog.V(5).Infof("refresh failed, will continue to login: %s", err)
 			} else {
 				tokenRefreshed = true
 			}
 
+			if token.RefreshToken == "" {
```

Comment:
do you happen to know under what condition `token.RefreshToken` would be empty? what was the auth flow being used?

Reply:
Hi @weinong !

I don't know the specific auth flow (if you're looking for the flow ID). We have our Kubernetes cluster setup to authenticate against Azure AD. One possible sticking point is that our Azure AD is setup to delegate to Google for SSO.

One thing we have been noticing for most users (created after January 2020, mostly) is that whenever their access token expires and is refreshed, the token that comes back has no refresh token field which makes it really difficult for people to do their work, as once that refreshed access token expires they need to do the device code sign-in again.

Reply:
@maxburke would it be possible to send me the fiddler trace showing the empty refresh token? I can engage with AAD team to take a look. you can send that to me at weinongw at microsoft.com :)

Reply:
@maxburke if you cannot send me the repro, can you at least add a log line such as
```
klog.V(10).Info("after token refresh refresh_token is still empty, reuse the existing refresh_token...")
```